FirebaseSwiftUI/FirebaseAuthSwiftUI/Sources/Views/MFAManagementView.swift

```diff
@@ -0,0 +1,235 @@
+    Binding(
+      get: { authService.passwordPrompt.isPromptingPassword },
+      set: { authService.passwordPrompt.isPromptingPassword = $0 }
+    )
+  }
+
+  public init() {}
+
+  private func loadEnrolledFactors() {
+    guard let user = authService.currentUser else { return }
+    enrolledFactors = user.multiFactor.enrolledFactors
+  }
+
+  private func unenrollFactor(_ factorUid: String) {
+    Task {
+      isLoading = true
+      errorMessage = ""
+
+      do {
+        let freshFactors = try await authService.unenrollMFA(factorUid)
+        enrolledFactors = freshFactors
+      } catch {
+        errorMessage = error.localizedDescription
+      }
+
+      isLoading = false
+    }
+  }
+
+  private func navigateToEnrollment() {
+    authService.authView = .mfaEnrollment
+  }
+
+  private func goBack() {
+    authService.authView = .authPicker
+  }
+}
+
+extension MFAManagementView: View {
+  public var body: some View {
+    VStack(spacing: 20) {
+      // Header with manual back button
+      HStack {
+        Button(action: {
+          authService.authView = .authPicker
+        }) {
+          HStack(spacing: 4) {
+            Image(systemName: "chevron.left")
+              .font(.system(size: 17, weight: .medium))
+            Text(authService.string.backButtonLabel)
+              .font(.system(size: 17))
+          }
+          .foregroundColor(.blue)
+        }
+        .accessibilityIdentifier("back-button")
+
+        Spacer()
+      }
+      .padding(.horizontal)
+
+      // Title section
+      VStack {
+        Text("Two-Factor Authentication")
+          .font(.largeTitle)
+          .fontWeight(.bold)
+          .multilineTextAlignment(.center)
+
+        Text("Manage your authentication methods")
+          .font(.subheadline)
+          .foregroundColor(.secondary)
+          .multilineTextAlignment(.center)
+      }
+      .padding(.horizontal)
+
+      if enrolledFactors.isEmpty {
+        // No factors enrolled
+        VStack(spacing: 16) {
+          Image(systemName: "shield.slash")
+            .font(.system(size: 48))
+            .foregroundColor(.orange)
+
+          Text("No Authentication Methods")
+            .font(.title2)
+            .fontWeight(.semibold)
+
+          Text(
+            "Set up two-factor authentication to add an extra layer of security to your account."
+          )
+          .font(.body)
+          .foregroundColor(.secondary)
+          .multilineTextAlignment(.center)
+          .padding(.horizontal)
+
+          Button("Set Up Two-Factor Authentication") {
+            navigateToEnrollment()
+          }
+          .buttonStyle(.borderedProminent)
+          .accessibilityIdentifier("setup-mfa-button")
+        }
+      } else {
+        // Show enrolled factors
+        VStack(alignment: .leading, spacing: 16) {
+          Text("Enrolled Methods")
+            .font(.headline)
+            .padding(.horizontal)
+
+          ForEach(enrolledFactors, id: \.uid) { factor in
```

Comment:
can MultiFactorInfo conform to Identifiable, where id is a computed property from uid?